### PR TITLE
Remove Docs.Bookmarks

### PR DESCRIPTION
**Pull request by hdgarrood**
We don't actually need the full Bookmarks to be able to generate links
between packages properly, and in fact all that Pursuit uses the
pkgBookmarks field for now is to work out which package it needs to link
to, given a module name. That is, all it needs is a Map ModuleName
PackageName. So we can simplify and speed up the JSON generation by
replacing the pkgBookmarks with such a Map.

This gets us one step closer to being able to get rid of
Language.PureScript.AST.Exported, refs #2130.

### CODE REVIEW

File: src/Language/PureScript/Docs/Convert.hs [hunk 1]
```diff
@@ -3,28 +3,27 @@
...
 module Language.PureScript.Docs.Convert
   ( convertModules
+  , convertModulesWithEnv
   , convertModulesInPackage
-  , collectBookmarks
+  , convertModulesInPackageWithEnv
   ) where
 
-import Prelude.Compat
+import Prelude (id, String)
```
Comment (kRITZCREEK) [inline src/Language/PureScript/Docs/Convert.hs]: Just in case you didn't know `Protolude` exports `id` as `identity` to free up the name `id`.

Comment (hdgarrood) [inline src/Language/PureScript/Docs/Convert.hs]: Ah right, thanks!

**Comment by hdgarrood** [comment]
CI failure is that intermittent issue with parsing HPC files again. I'm going to try clearing that job's cache.

**Comment by hdgarrood** [comment]
There's no cache for this PR appearing in the travis settings. :slightly_frowning_face: I guess we can look at it later. Maybe we should disable coverage for now, actually, since apparently the last time it worked was 6 months ago: https://coveralls.io/github/purescript/purescript

**Comment by hdgarrood** [comment]
Cool :) I think I'll wait a little bit and see if this CI issue sorts itself out before sending a PR disabling coverage.

**Pull request merged**